kaggle_environments/envs/halite/halite.py

```diff
@@ -308,18 +289,30 @@ def create_uid():
     # Detect collisions
     # 1. Ships into Foreign Shipyards.
     # 2. Ships into Ships.
-    board = [[-1, {}] for _ in range(size ** 2)]
+    board = [[-1, {}, -1] for _ in range(size ** 2)]
     for index, agent in enumerate(state):
         if agent.status != "ACTIVE":
             continue
         _, shipyards, ships = obs.players[index]
         for uid, shipyard_pos in shipyards.items():
             board[shipyard_pos][0] = index
+            board[shipyard_pos][2] = uid
         for uid, ship in ships.items():
             board[ship[0]][1][uid] = index
     for pos, cell in enumerate(board):
-        shipyard, ships = cell
-        # Detect Ship Collisions.
+        shipyard, ships, shipyard_uid = cell
+        # Detect Shipyard Collisions.
+        if shipyard > -1:
+            for uid, index in list(ships.items()):
+                if shipyard != index:
+                    print ("Collision with shipyard by", uid, shipyard_uid, shipyard, index)
+                    del ships[uid]
+                    del obs.players[index][2][uid]
+
+                    if feature_flags["shipyards destroyed on collision, prior to spawn v2"]:
```

Comment:
This probably shouldn't be here, unless we've got a feature_flags dictionary somewhere.

Reply:
Apart from that, looks good! Thanks Sam!

Reply:
Haha just noticed that, I pushed an update a minute ago so if you refresh it should be good now.